### PR TITLE
feat: support action in package.json

### PR DESCRIPTION
**Pull request by idoprz**
one of the tests didn't passed. After discussion with Ira, I think it was local problem. Let's check it before merging this PR

### CODE REVIEW

File: tests/basctlServer.spec.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ describe("basctlServer", () => {
             startBasctlServer()   
     });
 
-    it("startBasctlServer socket exists, " +
+    it.only("startBasctlServer socket exists, " +
```
Comment (tomer-epstein) [inline tests/basctlServer.spec.ts:96]: remove .only

File: tests/extension.spec.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ mockVscode(testVscode, "src/actions/performer.ts");
 mockVscode(testVscode, "src/actions/basctlServer.ts");
 import * as extension from "../src/extension";
 import * as performer from '../src/actions/performer';
-import * as basctlServer from '../src/actions/basctlServer';
+import * as basctlServer from '../src/basctlServer/basctlServer';
```
Comment (tomer-epstein) [inline tests/extension.spec.ts:19]: image

**Review by tomer-epstein — CHANGES_REQUESTED** [review]

File: tests/basctlServer.spec.ts (hunk 1 above)
Comment (idoprz) [reply tests/basctlServer.spec.ts]: fixed

File: src/actions/controller.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+
+export class ActionsController {
+    private readonly actions: IAction[] = [];
+
+    loadActions() {
+        this.getContributorExtension();
+    }
+
+    getContributorExtension() {
```
Comment (tomer-epstein) [inline src/actions/controller.ts:14]: getContributorExtensions or getContributorsActions

File: src/actions/controller.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  //     const extensionDependencies: string[] = _.get(extension, "packageJSON.extensionDependencies");
+  //     if (_.includes(extensionDependencies, "saposs.code-snippet")) {
+  //         if (contributorId === this.getExtensionId(extension)) {
+  //             return extension;
+  //         }
+  //     }
+
+  //     this.logger.warn(`Extension '${contributorId}' could not be found.`);
+  // });
+
```
Comment (tomer-epstein) [inline src/actions/controller.ts:52]: remove

**Review by tomer-epstein — CHANGES_REQUESTED** [review]

**Pull request merged**